Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -258,19 +321,42 @@ class SmartPage_Context
     }
   };
 
-  public SmartPage_PageRequestManager_endRequest = function (sender: unknown, args: Sys.WebForms.EndRequestEventArgs)
+  public OverrideErrorMessageForStatusCode (statusCode: number, message: string) : void {
+    let current = this._errorMessageMap.get(statusCode);
+    if (current == undefined) return;
+
+    current.errorMessage = message;
+  }
+
+  public SmartPage_PageRequestManager_endRequest = (sender: unknown, args: Sys.WebForms.EndRequestEventArgs) =>
   {
-    if (args.get_error() != undefined && (args.get_error() as any).httpStatusCode == '500')
+    if (args.get_error() != undefined)
     {
-      const errorMessage = args.get_error().message;
-      args.set_errorHandled(true);
+      let statusCode = (args.get_error() as any).httpStatusCode;
 
-      const errorBody = '<div class="SmartPageErrorBody"><div>' + errorMessage + '</div></div>';
+      let errorDeclaration = this._errorMessageMap.get(statusCode);
 
-      SmartPage_Context.Instance!.ShowMessage("SmartPageServerErrorMessage", errorBody);
+      if (errorDeclaration == undefined || statusCode == 200){
```

Comment:
Isn't the second part of the if superfluous since we don't have an entry for any of the 2xx HTTP codes?

Reply:
Comment in ticket hints that there can be a Statuscode 2xx with error, Idk how that would happen either, but well oh well

Reply:
My point was that since we don't have any 2xx status codes in the `_errorMessageMap` registered. So in this if `errorDeclaration == undefined` will already evaluate to `true` if `statusCode == 200` is `true`, making `statusCode == 200` superfluous.

Something I just saw since I was looking at the line again: In JS all values are either truthy or falsy so null/undefined checks are usually simplified to just `value` instead of  `value !== null` since `null` and `undefined` are both falsy anyway. There are some edge cases for certain falues (see falsy values https://developer.mozilla.org/en-US/docs/Glossary/Falsy), but for objects just `value` is ideal as it checks for `null` and `undefined :)